### PR TITLE
Disallow users to bypass BC breaking changes

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72314
* #72313

Differential Revision: [D33997712](https://our.internmc.facebook.com/intern/diff/D33997712)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33997712).

File: docs/guidance_for_upgraders.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Guidance for Operator Developer
+
+After you change either the operator signature in a BC breaking way or the semantics of the operator, you will need to write an “upgrader” to make the change non-BC breaking. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `
```
Comment (mruberry) [inline docs/guidance_for_upgraders.md:3]: This is not always true. Prototype and beta modules don't have BC-breaking guarantees and upgraders are only required if the operator is actually used by a torchscript or mobile model.

File: test/forward_backward_compatibility/check_forward_backward_compatibility.py [hunk 2]
```diff
@@ -227,8 +102,8 @@ def check_bc(existing_schemas):
     else:
         print(
             "The PR is introducing backward incompatible changes to the "
```
Comment (mruberry) [inline test/forward_backward_compatibility/check_forward_backward_compatibility.py:104]: Again we can't block developers from making BC-breaking changes in coordination with operators. This test also checks FC, too, which upgraders don't deal with.

File: docs/guidance_for_upgraders.md (hunk 1 above)
Comment (gmagogsfm) [reply docs/guidance_for_upgraders.md:3]: Ah, good to know. What is an example of such a module? Is there a canonical source of truth where we can query whether a module is beta and not dispatchable to TorchScript?

File: test/forward_backward_compatibility/check_forward_backward_compatibility.py (hunk 2 above)
Comment (gmagogsfm) [reply test/forward_backward_compatibility/check_forward_backward_compatibility.py:104]: What does "in coordination with operators" mean? The breaking change is on operator, so it is always in coordination with itself. 

Indeed, we can't deal with FC breaking changes now, it still has to follow 2 week wait policy, we we need at least an allowlist for FC.

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33997712).

**Comment by mruberry** [comment]
Per offline discussion we decided to think about our approach to testing BC/FC more generally before removing the allow list.

**Pull request closed without merging**